### PR TITLE
Remove Python 2.7 support

### PR DESCRIPTION
**Pull request by ColCarroll**
This drops support for python 2.7.

 [x] Remove all uses of six
 [x] Remove documentation saying we use python 2.7
 [x] Remove python 2.7 from testing grid

Open to suggestions for other easy places to remove support.

### CODE REVIEW

**Comment by junpenglao** [comment]
Should we also do `super(Normal, self).__init__` --> `super().__init__`?

**Comment by aloctavodia** [comment]
No need to inherit from `object` when declaring a class

**Comment by ColCarroll** [comment]
Thanks! Updated with both of those suggestions (and reminding myself how metaclasses are used in Python3).

File: pymc3/step_methods/slicer.py [hunk 1]
```diff
@@ -101,4 +101,3 @@ def competence(var, has_grad):
                 return Competence.PREFERRED
             return Competence.COMPATIBLE
         return Competence.INCOMPATIBLE
-
```
Comment (twiecki) [inline pymc3/step_methods/slicer.py:104]: new-line

File: pymc3/vartypes.py [hunk 2]
```diff
@@ -22,10 +20,7 @@
 continuous_types = float_types | complex_types
 discrete_types = bool_types | int_types
 
-if sys.version_info[0] == 3:
-    string_types = str
-else:
-    string_types = basestring
+string_types = str
```
Comment (twiecki) [inline pymc3/vartypes.py]: can probably just get rid of this.

**Comment by twiecki** [comment]
Love this, we can also:
* Get rid of joblib and all associated code
* Use `@` instead of `dot` for matrix multiplications

**Comment by ColCarroll** [comment]
Thanks for the check! Found that `2to2` has an `idioms` check that I used a little bit. Also got rid of the rest of the joblib hooks.  I've left off the matrix multiplication so far, and am also looking if pylint has anything interesting to say.

Removing the `WIP` -- I think I'm getting to places where we might want to review changes separately, and this seems ready to merge.

**Comment by twiecki** [comment]
I reviewed and it looks great.

**Comment by twiecki** [comment]
It's gone!

**Pull request merged**